components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/claims/impl/DefaultClaimHandler.java

```diff
@@ -779,6 +780,9 @@ protected Map<String, String> handleLocalClaims(String spStandardDialect,
                             appAssociatedRoles));
                 }
                 if (isAppRoleClaimRequested) {
+                    appAssociatedRoles = appAssociatedRoles.stream()
```

Comment:
Shall we handle this in federated flow also. Check `handleFederatedClaims` method

Reply:
addressed